compiler/AST/symbol.cpp

```diff
@@ -1853,6 +1853,22 @@ VarSymbol *new_RealSymbol(const char *n, IF1_float_type size) {
   return new_FloatSymbol(n, size, NUM_KIND_REAL, dtReal[size]);
 }
 
+VarSymbol *new_RealSymbol(float val) {
+  Immediate imm;
+  imm.v_float32 = val;
+  imm.const_kind = NUM_KIND_REAL;
+  imm.num_index = FLOAT_SIZE_32;
+  return new_ImmediateSymbol(&imm);
+}
+
+VarSymbol *new_RealSymbol(double val) {
```

Comment:
Minor, but could we either give these two different names (e.g. `new_Real32Symbol`) or adding a `IF1_float_type` argument (which would make it more similar to the int/uint versions).

Reply:
We could...  The IF1_float_type approach feels like overkill to me in that I don't think we'd want to permit a C double to become a real32 or a float a real64 (at least, not before having a realistic reason to do so), so it just feels like requiring additional information to be passed along needlessly in the current/common cases.  My interpretation of why new_[U]IntSymbol takes the size is because all [u]int symbols can be represented precisely / losslessly with the input type.

I'm more open to specializing the names, though it also feels (a) unnecessary to me given C++ overloading and (b) less aligned with new_[U]IntSymbol() in that those also take values rather than the strings of new_RealSymbol and new_imagSymbol.  So maybe let me know how strongly you feel about that approach.

Reply:
Yes, I agree about those things. I tend to not like to use overloading unless I have a good reason -- having different functions with the same name can make debugging harder and also make it harder to see which one is called at a glance. (and I don't think consistency with new_IntSymbol is a good enough reason here). But this is arguably a style issue and not a show stopper here by any means.

Reply:
OK.  To me this set of calls feels like an initializer / constructor call, which is a case where I tend to like overloading (e.g., I'd like to build a real literal out of this string or this float or this double.  Of course, that begs the question "Why is it not a constructor call?" but I don't know the answer to that offhand...